src/Microsoft.OData.Core/Json/JsonWriter.cs

```diff
@@ -487,11 +489,15 @@ public void EndTextWriterValueScope()
         /// </summary>
         public void Dispose()
         {
-            if (this.ArrayPool != null && this.buffer != null)
+            if (this.ArrayPool != null)
             {
-                BufferUtils.ReturnToBuffer(this.ArrayPool, this.buffer);
+                if (this.wrappedBuffer.Value != null)
+                {
+                    BufferUtils.ReturnToBuffer(this.ArrayPool, this.wrappedBuffer.Value);
+                    this.wrappedBuffer.Value = null;
+                }
+
                 this.ArrayPool = null;

```

Comment:
Could you explain the change in logic here?

In the original version this.ArrayPool is set to null only when the buffer is null. In the updated code, ArrayPool is set to null regardless of whether the buffer is null or not. Is this change significant? Or was it a bug in the original version?

Reply:
@habbes argh! I had rewritten this code when I was dealing with both a char buffer reference and a wrapped buffer. I'll rewrite it so the block is executed when both `this.ArrayPool` and `this.wrappedBuffer.Value` are `null`